File: Sem-3/DSA(C)/Selection_Sort.c
#include <stdio.h>
#include <conio.h>

int main()
{
    int len;
    printf("Enter the length of Array: ");
    scanf("%d", &len);
    int arr[len];

    printf("Enter the elements of the array: \n");
    for (int i = 0; i < len; i++)
    {
        scanf("%d", &arr[i]);
    }

    // ? SORTING
    selectionSort(arr, len);
    for (int i = 0; i < len; i++)
    {
        printf("%d, ", arr[i]);
    }
}

void selectionSort(int arr[], int len)
{
    int min, temp;
    for (int i = 0; i < len; i++)
    {
        min = i;
        for (int j = i + 1; j < len; j++)
        {
            if (arr[j] < arr[min])
            {
                min = j;
            }
            if (min != i)
            {
                temp = arr[i];
                arr[i] = arr[j];
                arr[j] = temp;
            }
        }
    }
}

File: Sem-3/Python/selection_sort.py
def selection_sort(arr):
    n = len(arr)
    for i in range(n):
        min_idx = i
        for j in range(i+1, n):
            if arr[j] < arr[min_idx]:
                min_idx = j
        arr[i], arr[min_idx] = arr[min_idx], arr[i]

# Example usage:
arr = []
num = int(input("Enter size of list : "))
print("Enter elements of list : ")
for n in range(num):
    a = int(input("-> "))
    arr.append(a)
selection_sort(arr)
print("Sorted array:", arr)